extension/src/setup/commands/index.ts

```diff
@@ -70,17 +99,223 @@ const addRemoteToProject = async (
+  }
+  return await Toast.showOutput(
+    internalCommands.executeCommand(
+      AvailableCommands.REMOTE,
+      dvcRoot,
+      SubCommand.MODIFY,
+      remote.config,
+      remote.name,
+      'url',
+      newUrl
+    )
+  )
+}
+
+const modifyRemote = async (
+  internalCommands: InternalCommands,
+  dvcRoot: string,
+  remote: RemoteWithConfig
+): Promise<void> => {
+  const option = await quickPickOne(
+    Object.values(ModifyOptions),
+    'Select an Option to Modify'
+  )
+
+  if (!option) {
+    return
+  }
+
+  if (option === ModifyOptions.NAME) {
+    return modifyRemoteName(internalCommands, dvcRoot, remote)
+  }
+
+  if (option === ModifyOptions.URL) {
+    return modifyRemoteUrl(internalCommands, dvcRoot, remote)
+  }
+}
+
+const collectFromRemoteList = (
+  acc: QuickPickItemWithValue<RemoteWithConfig>[],
+  remoteList: string | undefined,
+  config: typeof Flag.LOCAL | typeof Flag.PROJECT
+): void => {
+  for (const remote of trimAndSplit(remoteList ?? '')) {
+    const [name, url] = extractRemoteDetails(remote)
+    acc.push({
+      description: `(${config.slice(2)} config)`,
+      detail: url,
+      label: `${name}`,
+      value: { config, name }
+    })
+  }
+}
+
+const collectModifyItems = (
+  localRemoteList: string | undefined,
+  projectRemoteList: string | undefined
+): QuickPickItemWithValue<RemoteWithConfig>[] => {
+  const acc: QuickPickItemWithValue<RemoteWithConfig>[] = []
+  collectFromRemoteList(acc, localRemoteList, Flag.LOCAL)
+  collectFromRemoteList(acc, projectRemoteList, Flag.PROJECT)
+  return acc
+}
+
+export const pickRemoteAndModify = async (
+  internalCommands: InternalCommands,
+  dvcRoot: string
+): Promise<void> => {
+  const [localRemoteList, projectRemoteList] = await Promise.all([
+    internalCommands.executeCommand(
+      AvailableCommands.REMOTE,
+      dvcRoot,
+      SubCommand.LIST,
+      Flag.LOCAL
+    ),
+    internalCommands.executeCommand(
+      AvailableCommands.REMOTE,
+      dvcRoot,
+      SubCommand.LIST,
+      Flag.PROJECT
+    )
+  ])
+  const remotes = collectModifyItems(localRemoteList, projectRemoteList)
+
+  if (!definedAndNonEmpty(remotes)) {
+    return Toast.showError('No remotes to modify')
+  }
+
+  const remote = await getOnlyOrPickRemote(
+    remotes,
+    Title.SELECT_REMOTE_TO_MODIFY
+  )
+  if (!remote) {
+    return
+  }
+
+  return modifyRemote(internalCommands, dvcRoot, remote)
+}
+
+const collectRemoveItems = (remotes: string[]): QuickPickItemWithValue[] => {
+  const acc: QuickPickItemWithValue[] = []
+  for (const remote of remotes) {
+    const [name, url] = extractRemoteDetails(remote)
+    acc.push({ detail: url, label: name, value: name })
+  }
+  return acc
+}
+
+const confirmAndRemove = async (
+  internalCommands: InternalCommands,
+  dvcRoot: string,
+  remote: string
+): Promise<void> => {
+  const removalConfirmed = await Modal.warnOfConsequences(
+    'Are you sure you want to remove this remote? This could be IRREVERSIBLE!',
```

Comment:
Should we mention the fact that we're removing the remote both in the project and locally or is that obvious enough?

Reply:
I think this makes it pretty obvious that it is a destructive action.